File: app/src/main/java/com/fight/covid/dagger/AppModule.java
package com.fight.covid.dagger;

import android.content.Context;

import androidx.annotation.NonNull;

import com.fight.covid.FightCovidApplication;
import com.fight.covid.room.CountriesDao;
import com.fight.covid.room.CountriesRoomDatabase;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    private Context context;

    public AppModule(Context context) {
        this.context = context;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return context;
    }

    @NonNull
    @Provides
    CountriesDao getCountriesDao(){
        return  CountriesRoomDatabase.Companion.getDatabase(context).countriesDao();
    }
}

File: app/src/main/java/com/fight/covid/dagger/AppModule.kt
package com.fight.covid.dagger

import android.content.Context
import com.fight.covid.room.CountriesDao
import com.fight.covid.room.CountriesRoomDatabase.Companion.getDatabase
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ApplicationComponent
import dagger.hilt.android.qualifiers.ApplicationContext
import javax.inject.Singleton

@InstallIn(ApplicationComponent::class)
@Module
class AppModule {


    @Provides
    @Singleton
    fun providesCountriesDao(@ApplicationContext appContext: Context):CountriesDao{
        return getDatabase(appContext).countriesDao()
    }


}